packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useMappingsOperations.ts

```diff
@@ -104,31 +95,35 @@ export const useMappingsOperations = ({ iModelId, getAccessToken, mappingClient
   const extractionClient = useExtractionClient();
   const [errorMessage, setErrorMessage] = useState<string | undefined>(undefined);
   const [isTogglingExtraction, setIsTogglingExtraction] = useState<boolean>(false);
-  const [extractionIconData, setExtractionIconData] = useState<ExtractionIconData>({
-    iconStatus: "warning",
-    iconMessage: "",
-  });
+  const {extractionStatusIcon, extractionMessageData, setExtractionStatusIcon, setExtractionMessageData} = useExtractionStatusDataContext();
   const [showExtractionMessageModal, setShowExtractionMessageModal] = useState<boolean>(false);
-  const [extractionMessageData, setExtractionMessageData] = useState<ExtractionMessageData[]>([]);
+  const previousStatusIcon = useRef<ExtractionStatusData>();
```

Comment:
Could you explain why you need to know the `previousStatusIcon`?

Reply:
This is to check whether there is any status (not undefined) previously stored in context to decide whether we should call `fetchExtractionStatus` to retrieve the status again.